packages/component-library/src/components/table-and-list/mt-data-table/mt-data-table.vue

```diff
@@ -417,34 +425,36 @@
                     </template>
 
                     <template v-else>
-                      <!-- Use the correct renderer for the column -->
-                      <mt-data-table-number-renderer
-                        v-if="column.renderer === 'number'"
-                        :data="data"
-                        :column-definition="column"
-                        @click="$emit('open-details', data)"
-                      />
+                      <slot :name="`column-${column.property}`" v-bind="{ data, column }">
```

Comment:
I think this can be insecure because this allows also overriding existing renderer. Better would be an approach like this:
```html
<template v-else>
  <!-- Use the correct renderer for the column -->
  <mt-data-table-number-renderer
    v-if="column.renderer === 'number'"
    :data="data"
    :column-definition="column"
    @click="$emit('open-details', data)"
  />

  <mt-data-table-text-renderer
    v-else-if="column.renderer === 'text'"
    :data="data"
    :column-definition="column"
    @click="$emit('open-details', data)"
  />

  <mt-data-table-badge-renderer
    v-else-if="column.renderer === 'badge'"
    :data="data"
    :column-definition="column"
    @click="$emit('open-details', data)"
  />

  <mt-data-table-price-renderer
    v-else-if="column.renderer === 'price'"
    :data="data"
    :column-definition="column"
    @click="$emit('open-details', data)"
  />

  <slot name="custom-column-renderer" :v-bind="{ data, column }"></slot>
</template>
```

Then it is possible to add new, custom renderer which can be defined in the column definition. E.g.
```html
<template #custom-column-renderer>
    <my-custom-renderer v-if="column.renderer ==== 'my-custom-renderer' />
</template>
```

Reply:
I think we can add the v-else-if for slot also. so we can control all the current content.
```
<template v-else>
  <!-- Use the correct renderer for the column -->
  <mt-data-table-number-renderer
    v-if="column.renderer === 'number'"
    :data="data"
    :column-definition="column"
    @click="$emit('open-details', data)"
  />

  <mt-data-table-text-renderer
    v-else-if="column.renderer === 'text'"
    :data="data"
    :column-definition="column"
    @click="$emit('open-details', data)"
  />

  <mt-data-table-badge-renderer
    v-else-if="column.renderer === 'badge'"
    :data="data"
    :column-definition="column"
    @click="$emit('open-details', data)"
  />

  <mt-data-table-price-renderer
    v-else-if="column.renderer === 'price'"
    :data="data"
    :column-definition="column"
    @click="$emit('open-details', data)"
  />

  <slot v-else-if="column.renderer === 'custom'" name="custom-column-renderer" :v-bind="{ data, column }"></slot>
</template>
```

Reply:
should we use v-else for slot ?
```js
<slot v-else name="custom-column-renderer" v-bind="{ data, column }" />
```